### PR TITLE
Doc - Modify Dispatcher to use simplified thread-pool-config values, …

### PR DESCRIPTION
**Pull request by hhimanshu**
…as described by @ktoso in the the issue (#21572)

### CODE REVIEW

**Comment by hhimanshu** [comment]
@ktoso , let me know if this is what you meant from your comment on the issue. If I misunderstood something, let me know. Thanks

File: akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala [hunk 1]
```diff
@@ -78,12 +78,8 @@ object DispatcherDocSpec {
       executor = "thread-pool-executor"
       # Configuration for the thread pool
       thread-pool-executor {
-        # minimum number of threads to cap factor-based core number to
-        core-pool-size-min = 2
-        # No of core threads ... ceil(available processors * factor)
-        core-pool-size-factor = 2.0
-        # maximum number of threads to cap factor-based number to
-        core-pool-size-max = 10
+        // or in Akka 2.4.2+
```
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala]: The docs will be for 2.4.12 and onwards so no need for this comment

**Review by johanandren — CHANGES_REQUESTED** [review]

**Comment by johanandren** [comment]
Yes, that is the change Konrad was referring to. Thanks!

File: akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala [hunk 2]
```diff
@@ -78,12 +78,8 @@ object DispatcherDocSpec {
...
+        fixed-pool-size = 16
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala]: I don't understand this change. The documentation where this is included is about how to configure a dispatcher https://raw.githubusercontent.com/akka/akka/master/akka-docs/rst/scala/dispatchers.rst

The original example is valid. The `fixed-pool-size` is an alternative way of doing it when you don't care about the factor.

I suggest that we add another example to the docs, below the original one.

```
And here's another example that uses the "thread-pool-executor" with a fixed thread pool size:

.. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#fixed-pool-size-dispatcher-config
```

Note that we already have that example (used further down in the page).

**Review by patriknw — CHANGES_REQUESTED** [review]

File: akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala (hunk 2 above)
Comment (johanandren) [inline akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala]: Oh, I assumed there was general agreeing on it since the ticket is triaged. But you are right, maybe make the "simple" one the default and "here is a more flexible way to configure the number of threads based on the number of cores" for the original one? 

I'd guess the rationale for the original ticket was that people get the three values wrong all the time and misunderstand that it boils down to a single number @ktoso ?

Comment (patriknw) [inline akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala]: yes, we can place the fixed size example first, but for completeness we should also include the factor example.

We can write

... with a fixed thread pool size (e.g. for IO bound tasks):

... with a thread pool size based on number of cores (e.g. for CPU bound tasks):

**Comment by patriknw** [comment]
Ping @hhimanshu ?

**Comment by hhimanshu** [comment]
@patriknw , my apologies, I was busy with some other work. I will prioritize it now and get back on this. I may have further questions btw. Thanks

**Comment by patriknw** [comment]
thanks a lot

**Comment by hhimanshu** [comment]
@patriknw, I made the changes. Let me know if I missed out anything

**Comment by hhimanshu** [comment]
ping @patriknw, does this looks good?

**Review by johanandren — APPROVED** [review]

**Comment by hhimanshu** [comment]
Thank you

File: akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala [hunk 3]
```diff
@@ -78,12 +78,7 @@ object DispatcherDocSpec {
...
       # Configuration for the thread pool
       thread-pool-executor {
-        # minimum number of threads to cap factor-based core number to
-        core-pool-size-min = 2
-        # No of core threads ... ceil(available processors * factor)
-        core-pool-size-factor = 2.0
-        # maximum number of threads to cap factor-based number to
-        core-pool-size-max = 10
+        fixed-pool-size = 16
       }
```
Comment (patriknw) [inline akka-docs/rst/scala/code/docs/dispatcher/DispatcherDocSpec.scala]: Something like this is already defined in `//#fixed-pool-size-dispatcher-config`.
Revert this change.

File: akka-docs/rst/scala/dispatchers.rst [hunk 4]
```diff
@@ -153,6 +153,10 @@ And then using it:
 
 .. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#defining-fixed-pool-size-dispatcher
 
+Another example that uses the thread pool based on the number of cores (e.g. for CPU bound tasks)
+
+.. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#my-bounded-config
```
Comment (patriknw) [inline akka-docs/rst/scala/dispatchers.rst]: Use `#my-thread-pool-dispatcher-config` here (the reverted)

File: akka-docs/rst/scala/dispatchers.rst [hunk 5]
```diff
@@ -153,6 +153,10 @@ And then using it:
```
Comment (patriknw) [inline akka-docs/rst/scala/dispatchers.rst]: At line 45, use this as the example instead of the previous, i.e.

```
another example that uses the "thread-pool-executor":
 
 .. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#fixed-pool-size-dispatcher-config
```

File: akka-docs/rst/scala/dispatchers.rst [hunk 6]
```diff
@@ -153,6 +153,10 @@ And then using it:
 
 .. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#defining-fixed-pool-size-dispatcher
 
+Another example that uses the thread pool based on the number of cores (e.g. for CPU bound tasks)
+
+.. includecode:: ../scala/code/docs/dispatcher/DispatcherDocSpec.scala#my-bounded-config
+
```
Comment (patriknw) [inline akka-docs/rst/scala/dispatchers.rst]: then there is probably similar changes to be done in the Java documentation: akka-docs/rst/java/dispatchers.rst

**Review by patriknw — CHANGES_REQUESTED** [review]

**Comment by patriknw** [comment]
I have completed this in https://github.com/akka/akka/pull/22357
